crates/schema/src/auto_migrate.rs

```diff
@@ -0,0 +1,718 @@
+use crate::{def::*, error::PrettyAlgebraicType, identifier::Identifier};
+use spacetimedb_data_structures::error_stream::{CollectAllErrors, CombineErrors, ErrorStream};
+use spacetimedb_lib::db::raw_def::v9::TableType;
+use spacetimedb_sats::WithTypespace;
+
+pub type Result<T> = std::result::Result<T, ErrorStream<AutoMigrateError>>;
+
+/// A plan for an automatic migration.
+#[derive(Debug)]
+pub struct AutoMigratePlan<'def> {
+    /// The old database definition.
+    pub old: &'def ModuleDef,
+    /// The new database definition.
+    pub new: &'def ModuleDef,
+    /// The checks to perform before the automatic migration.
+    pub prechecks: Vec<AutoMigratePrecheck<'def>>,
+    /// The migration steps to perform.
+    /// Order should not matter, as the steps are independent.
+    pub steps: Vec<AutoMigrateStep<'def>>,
```

Comment:
Is it guaranteed to not matter, or does it just happen to be that way because of the types of automigration we do now?